src/reconcile/rec_visibility.c

```diff
@@ -717,16 +717,27 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
      * Set the flag if the selected tombstone is an out-of-order or mixed mode to an update. Based
      * on this flag, the caller functions perform the history store truncation for this key.
      */
-    if (tombstone != NULL &&
+    if (!is_hs_page && tombstone != NULL &&
       !F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS)) {
         upd = upd_select->upd;
         while (upd != NULL && upd->txnid == WT_TXN_ABORTED)
```

Comment:
This while loop is not required anymore.

Reply:
Good catch, fixed in 7807909b18e53c22a6dd9a9b2da4c8d6fa54687b